### PR TITLE
LPS-188818 - Keep pagination info when navigating tabs 

### PR DESCRIPTION
**Pull request by juanjofgliferay**
### Problem
In the FDS View component the user could change information using the different tabs available. When a user changes the pagination info , save it, navigate to other tab and come back to the Pagination one, the old information is still visible.
The same problem occurs in the Details tab.

When the FDS View first render, it request the **general view** data, that is passed as a prop to the different tabs. To trigger a new data request the only dependency is the viewId, that it is something that does not change among the tabs, so whenever a user changes the tab, is loading stale **general view** data. This is also happening with the other tabs, but don't have relevant data to update the UI.

### Proposed solution
The parent FDS View passes an update function to the components that need to update the view after saving the data.
In the current PR, I'm only passing the callback to the Details and Pagination tabs, but could be passed to all of them to be consistent. 

### Before
[before.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/415b2f81-b2bc-4a76-a9f7-f4540c96046f)

### After
[after.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/a18ca721-df08-434a-9201-a4701776cea8)

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Pagination.tsx [hunk 1]
```diff
@@ -132,6 +133,7 @@ function Pagination({
 				type: 'success',
 			});
 			setListOfItemsPerPage(itemsPerPage);
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Pagination.tsx:135]: Using the `updateFDSTabHandler` would make `setListOfItemsPerPage(itemsPerPage)` unneccesary

Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Pagination.tsx:135]: Yep, I think about it but forgot to change and test. Thx

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
@CarlosBrichete we'd need to forward this PR before Q2 milestone closing if possible. Is there any chance that relevant failing tests are related to the pull?

thank you again!
cc @markocikos

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by CarlosBrichete** [comment]
Hey @dsanz, the failures seem to be not related to this PR, LGTM 👍

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Details.tsx [hunk 2]
```diff
@@ -70,6 +71,8 @@ const Details = ({
 					fdsViewLabelRef.current?.value ?? ''
 				);
 			}
+
+			updateFDSTabHandler(responseJSON);
```
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Details.tsx]: @juanjofgliferay Sorry to bother you, let me correct myself: better name than `onUpdateFDSView` is `onFDSViewUpdate`. The last word in name should be the most significant.

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by juanjofgliferay** [comment]
ci:stop

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Details.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Details.tsx]: Just in time :smile:

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Comment by juanjofgliferay** [comment]
ci:forward

**Pull request closed without merging**